diff --git i/components/wifi.c w/components/wifi.c
index 8863252..fa1e043 100644
--- i/components/wifi.c
+++ w/components/wifi.c
@@ -58,6 +58,10 @@
 		sscanf(datastart + 1, " %*d   %d  %*d  %*d\t\t  %*d\t   "
 		       "%*d\t\t%*d\t\t %*d\t  %*d\t\t %*d", &cur);
 
+		if (cur == 0) {
+			return NULL;
+		}
+
 		/* 70 is the max of /proc/net/wireless */
 		return bprintf("%d", (int)((float)cur / 70 * 100));
 	}
diff --git i/slstatus.c w/slstatus.c
index 96fa5b6..1c244a8 100644
--- i/slstatus.c
+++ w/slstatus.c
@@ -85,6 +85,8 @@ main(int argc, char *argv[])
 		status[0] = '\0';
 		for (i = len = 0; i < LEN(args); i++) {
 			if (!(res = args[i].func(args[i].args))) {
+				if (!*unknown_str)
+					continue;
 				res = unknown_str;
 			}
 			if ((ret = esnprintf(status + len, sizeof(status) - len,
