diff --git i/dwm.c w/dwm.c
index 4465af1..0a3780c 100644
--- i/dwm.c
+++ w/dwm.c
@@ -696,8 +696,8 @@ void
 drawbar(Monitor *m)
 {
 	int x, w, sw = 0;
-	int boxs = drw->fonts->h / 9;
-	int boxw = drw->fonts->h / 6 + 2;
+	int boxs = drw->fonts->h / 7;
+	int boxw = drw->fonts->h / 4 + 2;
 	unsigned int i, occ = 0, urg = 0;
 	Client *c;
 
@@ -730,10 +730,10 @@ drawbar(Monitor *m)
 
 	if ((w = m->ww - sw - x) > bh) {
 		if (m->sel) {
-			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
+			int mid = (m->ww - TEXTW(m->sel->name)) / 2 - x + (drw->fonts->h * 2);
+			drw_text(drw, x, 0, w, bh, mid, m->sel->name, 0);
 			if (m->sel->isfloating)
-				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
+				drw_rect(drw, (m->ww - TEXTW(m->sel->name) + drw->fonts->h * 3) / 2, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
 			drw_setscheme(drw, scheme[SchemeNorm]);
 			drw_rect(drw, x, 0, w, bh, 1, 1);
